app/tag-management/annotation/annotation.component.ts

```diff
@@ -1,111 +1,189 @@
+import { Component, OnInit, OnDestroy, ViewChild, ElementRef, HostListener } from '@angular/core';
+import { DomSanitizer, SafeHtml } from '@angular/platform-browser';
 import { ToasterService } from 'angular2-toaster';
 import { TranslateService } from 'ng2-translate';
 
+import { AnnotationTag } from './annotation-tag.model';
+import { CanvasComponent } from './canvas/canvas.component';
 import { Tag } from '../tag.model';
 import { TagService } from '../tag.service';
 
 @Component({
   moduleId: module.id,
   selector: 'hip-annotation',
-  templateUrl: 'annotation.component.html'
+  templateUrl: 'annotation.component.html',
+  styles: [`
+    .center {
+      text-align: center;
+      margin-bottom: 10px;
+    }
+  `]
 })
 export class AnnotationComponent implements OnInit, OnDestroy {
-  mainMenu = new Array<{ label: string, entries: Tag[] }>();
+
+  @ViewChild('content') content: ElementRef;
+  @ViewChild(CanvasComponent) canvas: CanvasComponent;
+
+  @HostListener('window:resize', ['$event']) onResize() {
+    this.canvasHeight = this.content.nativeElement.parentElement.offsetHeight;
+    this.canvasWidth = this.content.nativeElement.parentElement.offsetWidth;
+    for(let tag of this.tagsInDocument) {
+      tag.redrawConnection(this.canvas)
+    }
+  }
+
+  mode = 'annotate';
+  mainMenu: { label: string, entries: Tag[] }[] = [];
   selectedTag = Tag.emptyTag();
+  lastTag: AnnotationTag = null;
+  canvasHeight = 0;
+  canvasWidth = 0;
+  followMouse = false;
+  lastElement: HTMLElement = undefined;
+
   private stylesheet: HTMLStyleElement;
   private tags: Tag[];
   private translatedResponse: string;
+  private tagCounter = 0;
+  private tagsInDocument: AnnotationTag[] = [];
+  private annotateContent: SafeHtml = '';
 
   constructor(private tagService: TagService,
               private toasterService: ToasterService,
-              private translateService: TranslateService) {}
+              private translateService: TranslateService,
+              private sanitizer: DomSanitizer) {
+  }
 
   ngOnInit() {
+    this.tagService.getAnnotateContent(12)
+      .then((result: string) => {
+        this.annotateContent = this.sanitizer.bypassSecurityTrustHtml(result);
+        setTimeout(() => this.initModel(),5);
+      });
+
+
     this.tagService.getAllTags()
       .then(
         (response: any) => {
-          this.tags = response.sort(this.tagAlphaCompare);
+          this.tags = response.sort((a: Tag, b:Tag) => this.tagAlphaCompare(a,b));
           this.buildMenu();
 
           // generate a stylesheet for annotations out of tag styles
           this.stylesheet = document.createElement('style');
           for (let tag of this.tags) {
-            this.stylesheet.innerHTML += `#text *[data-tag-id="${tag.id}"] { background-color: ${tag.style} }`;
+            this.stylesheet.innerHTML += `#text *[data-tag-model-id="${tag.id}"],md-menu *[data-tag-model-id="${tag.id}"] { background-color: ${tag.style} }`;
```

Comment:
__[tslint]__
* exceeds maximum line length of 140

<!-- Comment by [SideCI](https://sideci.com) -->